### PR TITLE
Make coursier-related tasks public

### PR DESCRIPTION
**Pull request by smarter**
This follows the discussion in
https://github.com/coursier/coursier/issues/1181.

### CODE REVIEW

**Review by eed3si9n — APPROVED** [review]
I was hoping that LM API abstraction is enough, but the implementation difference in resolution is leaking about especially around priorities involving inter-project vs Maven Central, so I'll agree to opening up the details on Coursier configuration.

**Comment by eatkins** [comment]
This seems ok given that all of these are actually in the shaded coursier package. I'd only be concerned if they were exposing classes in the actual coursier package.

File: main/src/main/scala/sbt/internal/LMCoursier.scala [hunk 1]
```diff
@@ -22,7 +22,7 @@ import sbt.internal.librarymanagement.{ CoursierArtifactsTasks, CoursierInputsTa
 import sbt.util.Logger
 import sbt.io.syntax._
 
-private[sbt] object LMCoursier {
+object LMCoursier {
```
Comment (dwijnand) [inline main/src/main/scala/sbt/internal/LMCoursier.scala:25]: Change package?

Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/LMCoursier.scala:25]: Yea. I was gonna follow up with some moving things around.

**Pull request merged**